_posts/2020-06-12-coding-with-kids.md

```diff
@@ -0,0 +1,195 @@
+
+![Sphero draw](/img/coding-with-kids/sphero_draw.jpg)
+
+Long story short. They are awesome, but still they are kind of gathering dust in our house. I do not know the actual reason, but maybe the reason could be that they are too complete for me. I have found out that I would really like to build the robot myself. In contrast, here is my old project built on top of Netduino. If you look those two Sphero robots and compare to this, you surely get what I meant. Maybe I am just missing an angle from where to approach the RVR since you could add things like robotic arm to it. Nevertheless I kind of would personally like to look into arduino/netduino/F7 meadow instead of Sphero when building robots, just to make the process more comprehensive, but at the same time it could be too hard for kids.
+
+![Netduino](/img/coding-with-kids/netduino.jpg)
+
+## Mastering the art of Python
+
+I needed to learn more Python to my work project, so I implemented Space Invader clone with `pygame` with a twist that I replaced all the aliens with my own face. My kids loved to choose dads that were trying to catch their spaceship. The natural move was then to ask if they wanted to learn about making software like this, and of course they wanted. 
+
+In overall Python is a really good language for learning programming. There is plenty of sources. It is supported in many situations, and it is simple to get started. It comes with much less boilerplate code than for example Java or C# would come, which helps kids to focus into the actual thing and not to be distracted about "important but not necessary" features of many other languages.  
+
+#### Using book to learn Python
+
+I considered how to learn programming when you already knew Scratch, PowerShell and Turtle Roy. I decided that I could go into more comprehensive direction and lend a book from library. As always with programming books, there were lots of words, and we were not interested on every of them. We cherrypicked example projects from there that were written in english and translated them with my kid to finnish. One of the examples was using `from turtle import *` to make a drawing robot that knew how to build a house, so you could command Python from commandline to provide exactly same house over and over again. That was a fun process, but the end result is rather boring. Instead, I will present you a piece software that was faster to implement and provided much more fun on many occasions: generating funny phrases. The idea is really simple. Make a few arrays of words like subjects, deeds and objects and mix them randomly to provide unexpected phrases. Use your family names in subjects to make it even funnier. Here is a hastily from finnish-to-english translated version of the implementation.
```

Comment:
into a more comprehensive


from the library

not interested in

written in English

to Finnish

from the commandline

exactly the same

I will present you with a piece of software

Reply:
not interested on every of them -> not interested in every one of them.
Here is a hastily from finnish-to-english translated version of the implementation. -> Here is a hastily translated version of the implementation from Finnish to English.